velox/functions/prestosql/Reduce.cpp

```diff
@@ -168,7 +172,10 @@ class ReduceFunction : public exec::VectorFunction {
           nullptr,
           &localResult);
     }
-    context.moveOrCopyResult(localResult, rows, result);
+    if (flatArray->rawNulls()) {
+      localResult->addNulls(flatArray->rawNulls(), rows);
+    }
+    context.moveOrCopyResult(localResult, *nonNullRows, result);
```

Comment:
this should use `rows` instead of nonNullRows

Reply:
Done.

Reply:
Seems if we use `*nonNullRows` here, we don't need change UT `finalSelection`.